src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutSubItem.cs

```diff
@@ -1,47 +1,882 @@
+			}
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: UpdateParentOwner - parentPresenter=0x%p.", this, parentPresenter));
+#endif // MFSI_DEBUG
+
+			if (parentPresenter != null)
+			{
+				ISubMenuOwner parentSubMenuOwner;
+				parentSubMenuOwner = (parentPresenter as IMenuPresenter).Owner;
+
+#if MFSI_DEBUG
+				IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: UpdateParentOwner - parentSubMenuOwner=0x%p.", this, parentSubMenuOwner));
+#endif // MFSI_DEBUG
+
+				if (parentSubMenuOwner != null)
+				{
+					((ISubMenuOwner)this).ParentOwner = parentSubMenuOwner;
+				}
+			}
+		}
+
+		// Set the popup open or close status for MenuFlyoutSubItem and ensure the
+		// focus to the current presenter.
+		void SetIsOpen(bool isOpen)
+		{
+			bool isOpened = false;
+
+			isOpened = m_tpPopup.IsOpen;
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: SetIsOpen isOpen=%d, isOpened=%d.", this, isOpen, isOpened));
+#endif // MFSI_DEBUG
+
+			if (isOpen != isOpened)
+			{
+				(m_tpPresenter as IMenuPresenter).Owner = isOpen ? this : null;
+
+				MenuFlyoutPresenter parentPresenter;
+				parentPresenter = GetParentMenuFlyoutPresenter();
+
+				if (parentPresenter != null)
+				{
+					(parentPresenter as IMenuPresenter).SubPresenter = isOpen ? m_tpPresenter as MenuFlyoutPresenter : null;
+
+					IMenu owningMenu;
+					owningMenu = (parentPresenter as IMenuPresenter).OwningMenu;
+
+					if (owningMenu != null)
+					{
+						(m_tpPresenter as IMenuPresenter).OwningMenu = isOpen ? owningMenu : null;
+					}
+
+					UpdateParentOwner(parentPresenter);
+				}
+
+				// UNO TODO
+				//VisualTree* visualTree = VisualTree.GetForElementNoRef(GetHandle());
+				//if (visualTree)
+				//{
+				//	// Put the popup on the same VisualTree as this flyout sub item to make sure it shows up in the right place
+				//	(CPopup*)(m_tpPopup as Popup.GetHandle()).SetAssociatedVisualTree(visualTree);
+				//}
+
+				// Set the popup open or close state
+				m_tpPopup.IsOpen = isOpen;
+
+				// Set the focus to the displayed sub menu presenter when MenuFlyoutSubItem is opened and
+				// set the focus back to the original sub item when the displayed sub menu presenter is closed.
+				if (isOpen)
+				{
+					// Set the focus to the displayed sub menu presenter to navigate the each sub items
+					m_tpPresenter.Focus(FocusState.Programmatic);
+
+					// UNO TODO
+					// (DependencyObject.SetFocusedElement(
+					// 	spPresenterAsDO as DependencyObject,
+					// 	xaml.FocusState_Programmatic,
+					// 	false /*animateIfBringIntoView*/,
+					// 	&focusUpdated));
+				}
+				else
+				{
+					// Set the focus to the sub menu item
+					this.Focus(FocusState.Programmatic);
+
+					// UNO TODO
+					//(DependencyObject.SetFocusedElement(
+					//	spThisAsDO as DependencyObject,
+					//	xaml.FocusState_Programmatic,
+					//	false /*animateIfBringIntoView*/,
+					//	&focusUpdated));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)